Lib/test/test_zlib.py

```diff
@@ -18,6 +19,35 @@
         hasattr(zlib.decompressobj(), "copy"),
         'requires Decompress.copy()')
 
+# bpo-46623: On s390x, when a hardware accelerator is used, using different
+# ways to compress data with zlib can produce different compressed data.
+# Simplified test_pair() code:
+#
+#   def func1(datasrc):
+#       return zlib.compress(datasrc)
+#
+#   def func2(datasrc)
+#       co = zlib.compressobj()
+#       x1 = co.compress(data)
+#       x2 = co.flush()
+#       return x1 + x2
+#
+# On s390x if zlib uses a hardware accelerator, func1() creates a single
+# "final" compressed block whereas func2() produces 3 compressed blocks (the
+# last one is a final block). On other platforms with no accelerator, func1()
+# and func2() produce the same compressed data made of a single (final)
+# compressed block.
+#
+# Only the compressed data is different, the decompression returns the original
+# data:
+#
+#   zlib.decompress(func1()) == zlib.decompress(func2()) == datasrc
+#
+# Make the assumption that s390x always has an accelerator to simplify the
+# skip condition. Don't skip on Windows which doesn't have os.uname().
+skip_on_s390x = unittest.skipIf(hasattr(os, 'uname') and os.uname().machine == 's390x',
```

Comment:
To implementing Petr's comment suggestion:
```suggestion
def not_s390x():
    return not (hasattr(os, 'uname') and os.uname().machine == 's390x')
```
and replace `@skip...`s below with `if not_s390x:` on `assert`s.